lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -14,6 +14,19 @@ const {
 const publicReadBuckets = process.env.ALLOW_PUBLIC_READ_BUCKETS
     ? process.env.ALLOW_PUBLIC_READ_BUCKETS.split(',') : [];
 
+const checkPrincipalResult = Object.freeze({
```

Comment:
we can document here that the enum value order matters for principal comparison (_findBestPrincipalMatch)